CLI/turbot.py

```diff
@@ -136,28 +139,67 @@ def parse_filename(file_path):
-    parser.add_argument('-p', "--payload", action='append', help='Paths to .s files to be used for obfuscation mode as payloads. This flag can be used multiple times'
+    parser.add_argument('-p', "--payload", action='append', help='Optional: Paths to .s files to be used for obfuscation mode as payloads. This flag can be used multiple times'
                             ' to specify multiple payloads to add. These files must be named for the function to be called inside them. i.e. [func_name].s', metavar='payload', dest='payloads')
-    parser.add_argument('-m', '--mode', action='store', default='lazy', choices=INJECT_MODES, help='The obfuscation mode. Ignored if -O is not present.\n(default: %(default)s)', dest='mode')
-    parser.add_argument('source', metavar='F', type=str, help='The dir where the file is located source/path/file.ext')
-    parser.add_argument('destination', metavar='T', type=str, help='The dir where the output file will be stored output/path/')
+    parser.add_argument('-m', '--mode', action='store', default='lazy', choices=INJECT_MODES, help='Optional: The obfuscation mode. Ignored if -O is not present.\n(default: %(default)s)', dest='mode')
+    parser.add_argument('source', nargs='?', default='', metavar='source', type=str, help='The dir where the file is located source/path/file.ext')
+    parser.add_argument('destination', nargs='?', default=os.getcwd(), metavar='destination', type=str, help='Optional: The dir where the output file will be stored "/output/path/filename-prefix.ext"' 
+                            ' DO NOT include file extension when "-A" flag is set as it is auto generated')
 
     args = parser.parse_args(sys.argv[1:]) # Ignores the initial sys.argv which contains the path to this script
 
-    # Parses filename from source dir and prepare names for created outputs.
+    if len(sys.argv)==1:
+        parser.print_help(sys.stderr)
+        sys.exit(1)
+    if not os.path.isfile(args.source):
+        parser.print_help(sys.stderr)
+        eprint('Error: Provided source argument is not a file')
+        sys.exit(1)
+    if not isValidDestination(args.destination):
+        parser.print_help(sys.stderr)
+        eprint('Error: Provided destination argument is not a directory')
+        sys.exit(1)
+
+    # Set the base file name for use when destination is not specified
     filename = parse_filename(args.source)
-    dis_dest = os.path.join(args.destination, f'{filename}-disassembled.s')
-    obf_dest = os.path.join(args.destination, f'{filename}-obfuscated.s')
-    rea_dest = os.path.join(args.destination, f'{filename}-reassembled.out')
 
-    print(args)
-    if args.D == args.R == args.O:     # Default behaviour, step through all commands (happens if all false or all true)
-        print('Running default command')
+    if args.A or (args.D and args.R and args.O):     # Step through all commands (happens if all false or all true)
```

Comment:
I assume you changed this so that it wouldn't start with `-DRO` is none specified? If so, please correct the comment.
Otherwise, it needs to be `args.D == args.R == args.O`
`false and false and false` = false
`false == false == false` = true

Reply:
So "args.A = true" is the same as "D = true and R = true and O = true", that's the expected outcome.

Reply:
My point is more that previously the turbot program would do all be default, but this change prevents this. I don't mind which is the desired outcome